src/util/union_find.cpp

```diff
@@ -51,11 +51,11 @@ void unsigned_union_find::isolate(size_type a)
     if(c==1)
       return;
 
-    assert(c>=2);
+    INVARIANT(c >= 2, "Ensure unreachability for c <= 1"); // TODO review
```

Comment:
So that's really to say that `c == 0` is impossible? Maybe that should be checked above, right after the assignment to `c`?